dalm/training/generator_only/trainer.py

```diff
@@ -0,0 +1,300 @@
+    parser.add_argument("--shuffle_buffer", type=int, default=5000, help="the shuffle buffer size")
+    parser.add_argument("--seq_length", type=int, default=2600, help="the sequence length")
+    parser.add_argument("--num_workers", type=int, default=4, help="the number of workers")
+
+    parser.add_argument("--eval_steps", type=int, default=200, help="the evaluation frequency")
+    parser.add_argument("--num_train_epochs", type=int, default=3, help="the number of training epochs")
+    parser.add_argument("--logging_steps", type=int, default=10, help="the logging frequency")
+    parser.add_argument("--per_device_train_batch_size", type=int, default=1, help="the per device train batch size")
+    parser.add_argument("--per_device_eval_batch_size", type=int, default=1, help="the per device eval batch size")
+    parser.add_argument("--gradient_accumulation_steps", type=int, default=32, help="the gradient accumulation steps")
+    parser.add_argument(
+        "--gradient_checkpointing", type=bool, default=True, help="whether to use gradient checkpointing"
+    )
+    parser.add_argument("--group_by_length", type=bool, default=False, help="whether to group by length")
+    parser.add_argument("--no-packing", action="store_true", help="whether to not pack the sequences")
+
+    parser.add_argument("--lora_alpha", type=float, default=512, help="the lora alpha parameter")
+    parser.add_argument("--lora_dropout", type=float, default=0.05, help="the lora dropout parameter")
+    parser.add_argument("--lora_r", type=int, default=256, help="the lora r parameter")
+
+    parser.add_argument("--learning_rate", type=float, default=1e-4, help="the learning rate")
+    parser.add_argument("--lr_scheduler_type", type=str, default="cosine", help="the lr scheduler type")
+    parser.add_argument("--num_warmup_steps", type=int, default=100, help="the number of warmup steps")
+    parser.add_argument("--weight_decay", type=float, default=0.05, help="weight decay")
+    parser.add_argument("--optimizer_type", type=str, default="paged_adamw_32bit", help="the optimizer type")
+
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="./generator_finetuned_model",
+        help="the output directory where the model will be saved",
+    )
+    parser.add_argument("--neftune_noise_alpha", type=int, default=5, help="the noise alpha for neftune")
+    parser.add_argument("--run_name", type=str, default="generator_finetuning", help="the tracker run name")
+    return parser.parse_args()
+
+
+def train_generator(
+    model_name: str,
+    dataset_name: str,
+    local_dataset: bool,
+    num_train_epochs: int,
+    split: str,
+    size_valid_set: Optional[int],
+    validation_split: Optional[float],
+    streaming: bool,
+    shuffle_buffer: Optional[int],
+    seq_length: int,
+    num_workers: int,
+    eval_steps: int,
+    logging_steps: int,
+    per_device_train_batch_size: int,
```

Comment:
It would make life for callers a lot easier if there were default values for most of the parameters of this function which matched the CLI defaults, because otherwise the caller will have to duplicate the default values.  If we need to change the defaults, we'll need to change them in two places which is error-prone.

Constants could be be introduced to avoid duplicating the values for the CLI default and the function default values.